pkg/componentreadiness/jiraautomator/jiraautomator.go

```
@@ -0,0 +1,597 @@
+
+				// find all the defined variantBasedComponentRegressionThreshold this column is relevant to
+				for k, n := range col.Variants {
+					v := Variant{Name: k, Value: n}
+					if threshold, ok := j.variantBasedComponentRegressionThreshold[v]; ok {
+						columnToVariantsToThreshold[columnID] = map[Variant]int{v: threshold}
+					}
+				}
+			}
+		}
+	}
+	for _, row := range report.Rows {
+		for _, col := range row.Columns {
+			columnKeyBytes, err := json.Marshal(col.ColumnIdentification)
+			if err != nil {
+				return componentRegressedTests, err
+			}
+			columnID := string(columnKeyBytes)
+			useVariantBasedComponent := false
+			if variantToThreshold, ok := columnToVariantsToThreshold[columnID]; ok {
+				for variant, threshold := range variantToThreshold {
+					if threshold < columnToRegressionCount[columnID] {
+						if component, ok := variantBasedComponents[variant]; ok {
+							componentRegressedTests[component] = append(componentRegressedTests[component], col.RegressedTests...)
+							useVariantBasedComponent = true
+						}
+					}
+				}
+			}
+			if !useVariantBasedComponent {
+				componentRegressedTests[row.Component] = append(componentRegressedTests[row.Component], col.RegressedTests...)
+			}
+		}
+	}
+	for component, tests := range componentRegressedTests {
+		sort.Slice(tests, func(i, j int) bool {
+			return tests[i].ReportStatus < tests[j].ReportStatus
+		})
+		componentRegressedTests[component] = tests
+	}
+	return componentRegressedTests, nil
+}
+
+func (j JiraAutomator) integrateJiraForView(view crtype.View) error {
+
+	logger := log.WithField("view", view.Name)
+	logger.Info("jira integration for view")
+
+	report, err := j.getComponentReportForView(view)
+	if err != nil {
+		logger.WithError(err).Error("error getting report for view")
+	}
+
+	componentRegressedTests, err := j.groupRegressedTestsByComponents(report)
+	if err != nil {
+		logger.WithError(err).Error("error getting regressed tests from report")
```

This will continue processing with a bad componentRegressedTests? Suspect you want to return the err here.